wazp/pages/04_dashboard.py

```diff
@@ -1,15 +1,228 @@
+# import base64
+# import pdb
+
+# import pathlib as pl
+# import pdb
+
 import dash
-from dash import html
+
+# import dash_bootstrap_components as dbc
+import pandas as pd
+import plotly.express as px
+
+# import yaml
+from dash import dcc, html  # dash_table
+
+# import wazp.utils as utils
 
 dash.register_page(__name__)
 
+
+##########################
+# Read dataframe for one h5 file---this will be part of the figs' callbacks
+h5_file_path = """/Users/sofia/Documents_local/Zoo_SWC project/WAZP/
+sample_project_2/pose_estimation_results/jwaspE_nectar-open-
+close_controlDLC_resnet50_jwasp_femaleandmaleSep12shuffle1_1000000.h5"""
+df_trajectories = pd.read_hdf(h5_file_path.replace("\n", ""))
+df_trajectories.columns = df_trajectories.columns.droplevel()
+
+
+########################
+# Prepare figures
+# Trajectories
+fig_trajectories = px.scatter(
+    df_trajectories["head"],
+    x="x",
+    y="y",
+    labels={
+        "x": "x-axis (px)",
+        "y": "y-axis (px)",
+        "likelihood": "likelihood",
+    },
+    color="likelihood",
+    custom_data=df_trajectories["head"].columns,
+    title="Raw trajectories",
+)
+fig_trajectories.update_layout(
+    clickmode="event+select",
+    # xaxis_range=[0,1300],
+    # yaxis_range=[0,1100]
+)
+fig_trajectories.update_yaxes(
+    scaleanchor="x",
+    scaleratio=1,
+)
+fig_trajectories.update_traces(marker_size=5)
+
+# Heatmap
+fig_heatmap = px.density_heatmap(
+    df_trajectories["head"],
+    x="x",
+    y="y",
+    labels={
+        "x": "x-axis (px)",
+        "y": "y-axis (px)",
+    },
+    nbinsx=10,
+    nbinsy=10,
+    title="Heatmap",
+)
+fig_heatmap.update_layout(
+    clickmode="event+select",
+    # xaxis_range=[0,1300],
+    # yaxis_range=[0,1200]
+)
+fig_heatmap.update_yaxes(
+    scaleanchor="x",
+    scaleratio=1,
+)
+
+# Barplot occupancy
+# counts, xedges, yedges, _ = plt.hist2d(
+#     data['head'].x.dropna(),
+#     data['head'].y.dropna(),
+#     bins=100)
+
+# mesh = np.meshgrid(xedges, yedges)
+
+# mask_top_centre = (mesh[0] > 400) & (mesh[0] < 600) & (mesh[1] > 500)
+# & (mesh[1] < 900)
+# 100*np.sum(mask_top_centre[:-1,:-1] * counts) / np.sum(counts)
+
+df_occupancy = pd.DataFrame(
+    {
+        "ROI": [
+            "ROI 1",
+            "ROI 2",
+            "ROI 3",
+            "ROI 4",
+            "ROI 1",
+            "ROI 2",
+            "ROI 3",
+            "ROI 4",
+        ],
+        "occupancy": [0.4, 0.2, 0.4, 0.5, 0.3, 0.8, 0.7, 0.1],
+        "sex": ["M", "M", "M", "M", "F", "F", "F", "F"],
+    }
+)
```

Comment:
Oh, I see your secrets.
Is the commented-out stuff an attempt to make the plot programmatically?